util/patricia_trie/src/triedbmut.rs

```diff
@@ -334,29 +331,35 @@ impl<'a> TrieDBMut<'a> {
 			root_handle: root_handle,
 			death_row: HashSet::new(),
 			hash_count: 0,
+			marker: PhantomData,
 		})
 	}
 	/// Get the backing database.
-	pub fn db(&self) -> &HashDB {
+	pub fn db(&self) -> &HashDB<H> {
 		self.db
 	}
 
 	/// Get the backing database mutably.
-	pub fn db_mut(&mut self) -> &mut HashDB {
+	pub fn db_mut(&mut self) -> &mut HashDB<H> {
 		self.db
 	}
 
 	// cache a node by hash
-	fn cache(&mut self, hash: H256) -> super::Result<StorageHandle> {
-		let node_rlp = self.db.get(&hash).ok_or_else(|| Box::new(TrieError::IncompleteDatabase(hash)))?;
-		let node = Node::from_rlp(&node_rlp, &*self.db, &mut self.storage);
+	fn cache(&mut self, hash: H::Out) -> Result<StorageHandle, H::Out, C::Error> {
+		let node_encoded = self.db.get(&hash).ok_or_else(|| TrieError::IncompleteDatabase(hash))?;
+		// let node_encoded = self.db.get(&hash).ok_or_else(|| Box::new(TrieError::IncompleteDatabase(hash)))?;
```

Comment:
obsolete?

Reply:
No this one's actually wrong, good catch.